tests/fixtures/event/base.py

```diff
@@ -0,0 +1,392 @@
+"""
+Base class definitions for the event fixture
+"""
+from faker import Faker
+
+# Faker.seed(0)
+FAKE = Faker()
+
+
+class BaseEventField:
+    """BaseEventField Interface.
+    The Event Field is just a wrapper around it's value which is obtained
+    by calling it's generator function.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        generator,
+        removed=False,
+        optional=False,
+        dependency=None,
+        nullable=False,
+        emptiable_str=False,
+        emptiable_dict=False,
```

Comment:
Question:
The fields `nullable, emptiable_str, emptiable_dict` are only used in the `FreeEventField` class.
I can move those to the `FreeEventField` class but will have to redefine the 
 `__add__, __radd__ and __getitem__` functions which results in more code to maintain.
Do you prefer if i keep those fields here or move them?

Reply:
Move them where they are useful! The base class should only contain the smallest common part.

Reply:
Thanks, will be done